### PR TITLE
Add 1inch price estimation

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

This PR adds 1inch API as part of the Legacy API price competition.

image

# Scope

## Only Sell Orders
This will query only for SELL ORDERS, I'm not aware of the API to query a BUY order

## Issue with Sell Orders
I noticed there's a bug which makes the app to select the order Kind to BUY Order when it is in fact a SELL Order.

Any time I modify the sell amount, should be making the order a SELL Order. I see a weird behaviour

# To Test

> Elena, it might be worth waiting and not testing this until the issue described in the "Scope" part is solved.

Check the requests. It should be querying 1inch price.

If 1inch price is better, it should win the price competition and display that price. 

This should be tested in Gnosis Chain and Mainnet. Also, we should make sure the ETH estimation works too.

### CODE REVIEW

File: src/cow-react/api/1inch/index.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+  // Buy/sell token and side (sell/buy)
+  const { sellToken, buyToken } = getTokensFromMarket({ baseToken, quoteToken, kind })
+
+  if (OrderKind.BUY) {
+    // API doesn't support buy orders
+    return null
+  }
+
+  const priceProise = _get(chainId, `/price?fromTokenAddress=${sellToken}&toTokenAddress=${buyToken}&amount=${amount}`)
```
Comment (shoom3301) [inline src/cow-react/api/1inch/index.ts:109]: https://api.1inch.io/v5.0/1/price?fromTokenAddress=0x111111111117dc0aa78b770fa6a738034120c302&toTokenAddress=0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2&amount=1000000000000000000

Is the URL valid? I got 404

Comment (anxolin) [reply src/cow-react/api/1inch/index.ts:109]: yes, fix on its way, i pushed before testing. Now is working, but need to push changes.

One question, i don't see how to do a BUY order. Do they support that?

Comment (shoom3301) [reply src/cow-react/api/1inch/index.ts]: They don't :(

File: src/cow-react/api/1inch/index.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  // Buy/sell token and side (sell/buy)
+  const { sellToken, buyToken } = getTokensFromMarket({ baseToken, quoteToken, kind })
+
+  if (params.kind === OrderKind.BUY) {
+    // API doesn't support buy orders
+    console.debug(`[pricesApi:${API_NAME}] ${API_NAME} API don't support BUY Orders`, params)
+    return null
+  }
+
+  const priceProise = _get(chainId, `/quote?fromTokenAddress=${sellToken}&toTokenAddress=${buyToken}&amount=${amount}`)
```
Comment (shoom3301) [inline src/cow-react/api/1inch/index.ts:114]: Typo: `pricePromise`

**Review by shoom3301 — APPROVED** [review]

**Comment by anxolin** [comment]
Merging already. @elena-zh right now cannot be tested. We will need to test by enabling the LEGACY mode

**Pull request merged**